docs/schema.md

```diff
@@ -921,6 +921,70 @@ If this section is not specified, users see a warning when deploying the app to
 an Istio-enabled environment. The [`ISTIO_ENABLED`](#type-istio_enabled) type
 indicates whether Istio is enabled on the cluster.
 
+### assistedClusterCreation
+
+Use `assistedClusterCreation` to specify constraints for cluster creation as part of app deployment. These constraints determine whether a new cluster creation is allowed as part of deployment, and allow declaration of a specific node configuration for creation.
+
+`assistedClusterCreation` is optional if cluster creation is enabled. You can use it to specify a specific node configuration, but otherwise one will be generated according to defaults or specified `clusterConstraints`, if declared. The node config defined here will override the one suggested by `clusterConstraints`.
+
+```yaml
+properties:
+  # Property definitions...
+required:
+  # Required properties...
+x-google-marketplace:
+  clusterConstraints:
+    assistedClusterCreation:
+      type: DISABLED | STRICT
+      creationGuidance: #string
+      gke:
+        nodePool:
+        - numNodes: #integer
+          machineType: #string
+```
+
+Supported types:
+- `DISABLED`: The app doesn't support creating a new cluster as part of app deployment. User must select a pre-existing cluster.
+- `STRICT`: The app specified strict requirements for number of nodes and machine type which should be used if creating a new cluster as part of app deployment. 
+
+Use `creationGuidance` to specify instructions which will explain to end-user how they should create a cluster that will be compatible with the app. `creationGuidance` is required if constraint type is set as `DISABLED`.
+
+For instance, the following specifies that a new cluster can not be created as part of deployment and customer must select an existing cluster with GPU.
```

Comment:
" and customer must select an existing cluster with GPU." ->
" and the user must create a new cluster with the Ubuntu node image."

Reply:
Done.